warehouse/query-core/src/main/java/datawave/query/util/regex/RegexTrie.java

```diff
@@ -0,0 +1,159 @@
+package datawave.query.util.regex;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class can be used to turn a large list of strings into a regex.
+ * This may be used to handle queries that arrive with lists so large that the term limit is reached.
+ */
+public class RegexTrie {
+    final Node root = new Node();
+    class Node {
+        final HashMap<Character, Node> next = new HashMap<>();
+        boolean terminator = false;
+    }
+
+    public RegexTrie() {
+    }
+
+    public RegexTrie(List<String> strings) {
+        addAll(strings);
+    }
+
+    /**
+     * Add a set of strings into the trie
+     * @param strings A list of strings. May contain the empty string.
+     */
+    public void addAll(List<String> strings) {
+        for (String string : strings) {
+            add(string);
+        }
+    }
+
+    /**
+     * Add a string into the trie
+     * @param string A string.  May be the empty string.
+     */
+    public void add(String string) {
```

Comment:
Just a stylistic comment, but using `Map.computeIfAbsent`, I think you could simplify the add method to:
```java
public void add(String string) {
    Node current = root;
    for (int i = 0; i < string.length(); i++) {
        current = current.next.computeIfAbsent(string.charAt(i), n -> new Node());
    }
    current.terminator = true;
}
```